### PR TITLE
feat(test-runner): add GitHub Summary to Reporter

### LINKED ISSUE #14774: [Feature] GitHub Test Reporter Summary View
https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/

### PR DESCRIPTION
**Pull request by rwoll**
By default, the built-in GitHub reporter now generates a GitHub Action Summary[^1] to pretty-print a summary of problematic specs. (It's only included if there are flaky or failing tests. If everything passes, we don't add a summary report.)

This allows you to have a quick, (mostly) centralized glance at which specs are failing in GitHub across several jobs—including looking at the error! (Each Workflow has its own Summary page, and within that page multiple jobs are displayed—but unfortunately not aggregated into one table.)

For example, you can go to the [_Summary_ tab](https://github.com/rwoll/gha-summary-demo/actions/runs/2478038037)[^2] to see all the reports:

Screen Shot 2022-06-10 at 7 38 06 PM

And even see the errors 💥 (which are hidden by default to make scrolling easier):

Screen Shot 2022-06-10 at 7 38 21 PM

Previously, you had to click into each job and scroll through logs of each job to see this info.

This feature (and the already existing GH annotation feature) can be toggled off via a reporter option.

Resolves #14774.

[^1]: See https://github.blog/2022-05-09-supercharging-github-actions-with-job-summaries/ for the feature announcement! 🎉 
[^2]: Requires being authenticated to GitHub.

### CODE REVIEW

File: packages/playwright-test/src/reporters/github.ts [hunk 1]
```diff
@@ -57,20 +60,52 @@ class GitHubLogger {
...
+      `  <summary>${summaryTitleElement(t)}</summary>`,
+      `  <pre>${escapeHTML(stripAnsiEscapes(formatFailure(this.config, t).message))}</pre>`,
+      `</details>`,
+    ].join('\n'));
+
+    await fs.promises.appendFile(GITHUB_STEP_SUMMARY_PATH, [
+      summaryBadgeElement(allCases),
+      ...detailElements,
+      '',
+    ].join('\n')).catch(e => this._githubLogger.warning(`Playwright Test GitHub Reporter failed to write GitHub Summary: ${e}`));
```
Comment (rwoll) [inline packages/playwright-test/src/reporters/github.ts:108]: Move this catch up onto the callsite instead: https://github.com/microsoft/playwright/pull/14763/files#diff-22c9962e5f77e21b1ab9611b86c2ca7512794d31dced527e6680d275a596bd1aR80

**Comment by nil1511** [comment]
Is there any reason for closing this pr?

**Comment by rwoll** [comment]
> Is there any reason for closing this pr?

@nil1511 We closed it since—while it's helpful—the experience wasn't quite where we wanted it, and we need more time to think about how to present this information. In an ideal world, we'd get the reports from across the jobs aggregated into one summary instead of several.

**Comment by p0fi** [comment]
I understand that you closed it since it was not where you wanted it. But is there any other Issue or PR which tracks the new target experience? I would really love to use the job summary feature of GitHub actions instead of the annotations.

**Pull request closed without merging**